### PR TITLE
Core/Spells: POWER_BURN should not cancel when target isn't caster

### PR DESCRIPTION
**Pull request by matanshukry**
**Changes proposed:**

-  POWER_BURN should not cancel when target isn't caster

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

**Tests performed:**

(Does it build, tested in-game, etc.)

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Damage isn't applied at the moment. `Spell::EffectPowerBurn` will call `effectInfo->CalcValueMultiplier`, but it uses amplitutde which is 0 in the case of Ferocious Bite. Even if it wasn't, the result is multiplied by the amount of energy taken (25), rather than by the amount of damage dealt.

Not directly related (I might just open different issues for them):
- [ ] DisplayPower: seems to be like a single power that is displayed but you can have multiple such as energy + combopoints. Perhps we should convert it to a list rather than single? We could also use GetMaxPower(power) > 0, I believe.

### CODE REVIEW

**Comment by Robingadko** [comment]
>  Incorrect match between server and client energy.

Looks like there's already an issue #24677 for that. The same problem also for hunter focus and pet powers.

**Comment by matanshukry** [comment]
> > Incorrect match between server and client energy.
> 
> Looks like there's already an issue #24677 for that. The same problem also for hunter focus and pet powers.

Cool, I'll remove it from th todo list here and take a closer look at that issue then. Thanks!

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -534,14 +534,6 @@ void Spell::EffectSchoolDMG(SpellEffIndex effIndex)
             }
             case SPELLFAMILY_DRUID:
             {
-                // Ferocious Bite
-                if (m_caster->GetTypeId() == TYPEID_PLAYER && m_spellInfo->SpellFamilyFlags[3] & 0x1000)
```
Comment (Shauren) [inline src/server/game/Spells/SpellEffects.cpp:538]: Belongs in different PR

File: src/server/game/Spells/Spell.cpp [hunk 2]
```diff
@@ -5380,7 +5380,15 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
             case SPELL_EFFECT_POWER_DRAIN:
             {
                 // Can be area effect, Check only for players and not check if target - caster (spell can have multiply drain/burn effects)
-                if (m_caster->GetTypeId() == TYPEID_PLAYER)
```
Comment (Shauren) [inline src/server/game/Spells/Spell.cpp:5383]: Just completely remove all of it for both SPELL_EFFECT_POWER_BURN and SPELL_EFFECT_POWER_DRAIN

**Review by Shauren — CHANGES_REQUESTED** [review]

File: src/server/game/Spells/SpellEffects.cpp (hunk 1 above)
Comment (matanshukry) [reply src/server/game/Spells/SpellEffects.cpp:538]: Is it? the idea was this code depends on the fact we're not handling the power_burn, and this PR fixed it so the code isn't needed anymore.
no?

Comment (Shauren) [reply src/server/game/Spells/SpellEffects.cpp:538]: To me this belongs to ferocious bite PR

File: src/server/game/Spells/Spell.cpp (hunk 2 above)
Comment (Shauren) [reply src/server/game/Spells/Spell.cpp:5383]: Oh, I wanted you to just remove the entire switch case for those two effects
